### DISCUSSION TITLE
A/B redundancy

### DISCUSSION THREAD

**Discussion by enhany** [OE4T/meta-tegra#1269] [Q&A] [answered]
Hello,

is there easy way to create A/B rootfs redundancy without included mender?

**Comment by dwalkes** [comment]
Hi,
If you are talking about Jetpack 4 branches, you should be able to use `tegrademo-mender` at https://github.com/OE4T/tegra-demo-distro/tree/kirkstone-l4t-r32.7.x as an example and modify to use an update framework of your choice.  Others have used [swupdate](https://github.com/search?q=repo%3AOE4T%2Fmeta-tegra+swupdate&type=discussions), [RAUC](https://github.com/search?q=repo%3AOE4T%2Fmeta-tegra+rauc&type=discussions), etc and you can find some references to examples in some of the meeting notes and wiki pages.

If you are talking about Jetpack 5 branches, there's no working example yet as there is with Jetpack 4 but there are some basic building blocks available including capsule update and A/B update from NVIDIA.  Related discussions in the channel at https://matrix.to/#/#OE4T_jetpack-5-device-update:gitter.im if you'd like to follow along/contribute.

**Reply by enhany** [reply]
Hi,
thank you, yes, it's for Jetpack 4 branch.

I want to use A/B just for fail-over, not for updates. So need to crop all Mender staff from tegrademo-mender is the best option?

**Reply by dwalkes** [reply, marked as answer]
> I want to use A/B just for fail-over, not for updates

Got it

> So need to crop all Mender staff from tegrademo-mender is the best option?

Not necessarily - I think all you'll need to do is define a custom partition with both a/b slots included, you can use https://github.com/mendersoftware/meta-mender-community/tree/kirkstone/meta-mender-tegra/recipes-bsp/tegra-binaries/mender-custom-flash-layout and/or https://github.com/OE4T/meta-tegra/discussions/858 as a guide.

Then you just need to enable redundancy using one of the methods in https://docs.nvidia.com/jetson/archives/r34.1/DeveloperGuide/text/SD/Bootloader/UpdateAndRedundancy.html#enabling-bootloader-redundancy - see possibly related discussion at https://github.com/OE4T/meta-tegra/discussions/1267

So you might want to use the tegra demo distro and tegrademo-mender as a working example and a reference for troubleshooting issues with your custom one built without these layer(s) and the extra mender dependencies.

**Reply by enhany** [reply]
It works, thank you.
